src/extension/background-script/connectors/alby.ts

```diff
@@ -285,28 +283,60 @@ export default class Alby implements Connector {
       });
 
       authUrl += "&webln=false"; // stop getalby.com login modal launching lnurl auth
-      const authResult = await this.launchWebAuthFlow(authUrl);
-      const code = new URL(authResult).searchParams.get("code");
-      if (!code) {
-        throw new Error("Authentication failed: missing authResult");
-      }
 
-      const token = await authClient.requestAccessToken(code);
-      await this._updateOAuthToken(token.token);
-      return authClient;
+      const oAuthTab = await browser.tabs.create({ url: authUrl });
+
+      return new Promise<auth.OAuth2User>((resolve, reject) => {
+        const handleUpdated = (
+          tabId: number,
+          changeInfo: browser.Tabs.OnUpdatedChangeInfoType,
+          tab: browser.Tabs.Tab
+        ) => {
+          if (changeInfo.status === "complete" && tabId === oAuthTab.id) {
+            const authorizationCode = this.extractCodeFromTabUrl(tab.url);
+
+            if (authorizationCode) {
+              authClient
+                .requestAccessToken(authorizationCode)
+                .then((token) => {
+                  this._updateOAuthToken(token.token);
+                  resolve(authClient);
+                })
+                .catch((error) => {
+                  console.error("Failed to request new auth token", error);
+                  reject(error);
+                })
+                .finally(() => {
+                  browser.tabs.remove(tabId);
+                  browser.tabs.onUpdated.removeListener(handleUpdated);
+                });
+            }
+          }
+        };
+        const handleRemoved = (tabId: number) => {
```

Comment:
can you rename to `handleTabUpdated` and `handleTabRemoved`

Reply:
resolved